hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieUnMergedLogRecordScanner.java

```diff
@@ -83,7 +83,7 @@ protected <T> void processNextRecord(HoodieRecord<T> hoodieRecord) throws Except
 
   @Override
   protected void processNextDeletedRecord(DeleteRecord deleteRecord) {
-    throw new IllegalStateException("Not expected to see delete records in this log-scan mode. Check Job Config");
+    // no - op
```

Comment:
This seems irrelevant?

Reply:
create a follow up to track and expose the delete record keys

Reply:
Let's mention the Jira ticket here.

Reply:
There is no consumer of delete record key at the moment. Ideally I would defer this until we have a real necessity. When there is a demand, we will anyways add the functionality. I have created a follow up https://issues.apache.org/jira/browse/HUDI-8444 anyways.